pkg/syncer/storagepool/storageclasswatch.go

```diff
@@ -291,20 +292,95 @@ func (w *StorageClassWatch) addStorageClassPolicyAnnotations(ctx context.Context
 		log.Debugf("No storage policies to fetch")
 		return nil
 	}
-	profiles, err := w.vc.PbmRetrieveContent(ctx, w.policyIds)
+
+	profiles, err := w.fetchPolicies(ctx)
 	if err != nil {
-		log.Errorf("Failed to retrieve policy content. err: %v", err)
+		log.Errorf("fetchPolicies failed. err: %v", err)
 		return err
 	}
 	for _, profile := range profiles {
-		err = w.addStorageClassPolicyAnnotation(ctx, profile)
+		err := w.addStorageClassPolicyAnnotation(ctx, profile)
 		if err != nil {
 			log.Errorf("addStorageClassPolicyAnnotation failed. err: %v", err)
 		}
 	}
 	return nil
 }
 
+// fetchPolicies for known valid policyIDs from SPBM. The PbmRetrieveContent() is a batch API that returns the profile
+// contents for given profileIds. If any of the input profile IDs are invalid, then it returns an empty list of profiles
+// and an error that lists all the invalid profile IDs. In such a case, this function removes invalid profileIds from
+// the input and makes a second call to PbmRetrieveContent() with just the valid ones.
+func (w *StorageClassWatch) fetchPolicies(ctx context.Context) ([]cnsvsphere.SpbmPolicyContent, error) {
+	log := logger.GetLogger(ctx)
+	var profiles []cnsvsphere.SpbmPolicyContent
+	var err error
+	policyIds := w.policyIds
+	for i := 0; i < 2; i++ {
```

Comment:
Why do we attempt to retrieve policies in the loop? why  `i < 2` ? Can you put some notes regarding this logic?

Reply:
The first call might return an error with any invalid profileIDs found in the input. Those are removed from the input and a second call is made to the API to fetch the profile contents of valid ones. I have explained this in the function documentation above. Let me know if that can be improved.